homie-controller/src/values.rs

```diff
@@ -4,19 +4,36 @@ use std::num::ParseIntError;
 use std::str::FromStr;
 use thiserror::Error;
 
+/// An error encountered while parsing the value or format of a property.
 #[derive(Clone, Debug, Error, Eq, PartialEq)]
 pub enum ValueError {
+    /// The value of the property or attribute is not yet known, or not set by the device.
     #[error("Value not yet known.")]
     Unknown,
+    /// The method call expected the property to have a particular datatype, but the datatype sent
+    /// by the device was something different.
     #[error("Expected value of type {expected} but was {actual}.")]
     WrongDatatype {
+        /// The datatype expected by the method call.
         expected: Datatype,
+        /// The actual datatype of the property, as sent by the device.
         actual: Datatype,
     },
+    /// The format of the property couldn't be parsed or didn't match what was expected by the
+    /// method call.
```

Comment:
(for another day): is there any way to split these two cases, and/or show "what was expected" to help with debugging?

Reply:
I originally had two different enum variants, but I'm not convinced it's a useful distinction.

As for "what was expected", there's not necessarily an exact expected value. If you call `value::<ColorRGB>` we expect the string "rgb", but if you call `enum_values` or `range` then we just expect it to follow the right format.